snapshots/overlay/overlay_test.go

```diff
@@ -344,26 +339,29 @@ func testOverlayView(t *testing.T, newSnapshotter testsuite.SnapshotterFunc) {
 	if m.Source != "overlay" {
 		t.Errorf("mount source should be overlay but received %q", m.Source)
 	}
-
-	expectedOptions := 2
-	userxattr, err := NeedsUserXAttr(root)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if userxattr {
-		expectedOptions++
-	}
-
-	if len(m.Options) != expectedOptions {
-		t.Errorf("expected %d additional mount option but got %d", expectedOptions, len(m.Options))
-	}
-	lowers := getParents(ctx, o, root, "/tmp/view2")
-	expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
-	optIdx := 1
-	if userxattr {
-		optIdx++
+	indexOffOnce.Do(getIndexOff)
+	if indexOff {
+		if len(m.Options) != 2 {
+			t.Errorf("expected 1 additional mount option but got %d", len(m.Options))
+		}
+		lowers := getParents(ctx, o, root, "/tmp/view2")
+		expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
+		if m.Options[1] != expected {
+			t.Errorf("expected option %q but received %q", expected, m.Options[0])
+		}
+	} else {
+		if len(m.Options) != 1 {
+			t.Errorf("expected 1 additional mount option but got %d", len(m.Options))
+		}
+		lowers := getParents(ctx, o, root, "/tmp/view2")
+		expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
+		if m.Options[0] != expected {
+			t.Errorf("expected option %q but received %q", expected, m.Options[0])
+		}
 	}
-	if m.Options[optIdx] != expected {
-		t.Errorf("expected option %q but received %q", expected, m.Options[optIdx])
+}
+func getIndexOff() {
+	if _, err := os.Stat("/sys/module/overlay/parameters/index"); err == nil {
+		indexOff = true
 	}
 }
```

Comment:
The same check is present in the snapshotted itself; https://github.com/containerd/containerd/blob/ba8f9845ecefb9003d53322d4d5e607cca8c37bc/snapshots/overlay/overlay.go#L95-L99

I think it would be good to

- move that code to the `overlayutils` package; https://github.com/containerd/containerd/blob/ba8f9845ecefb9003d53322d4d5e607cca8c37bc/snapshots/overlay/overlayutils/check.go
- can add a sync.Once, but as suggested above, make that part of the function, using a similar approach as `RunningInUserNS() ` https://github.com/containerd/containerd/blob/717dde3c27145dab133a35250434b270935f834f/pkg/userns/userns_linux.go#L26-L34